From 3a6ca0192fdb6d22dba83851c93d0ec905bd0b38 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Tue, 22 Jan 2013 17:30:37 +0800
Subject: [PATCH 153/602] ENGR00241201 imx6: use more thread on FUSE daemon to
 make it faster.

make fuse daemon use more thread to make it faster.

Here is a Antutu IO data matrix on iMX6Q:

---------------------------------------------------------------------
Storage    |   2 thread (default)   | 4 Thread      |    8 Thread   |
---------------------------------------------------------------------
NAND write |     4.9M/s            | 5.2M/s        |      5.2M/s    |
eMMC write |     7.5M/s            | 8.8M/s        |      9.1M/s    |
---------------------------------------------------------------------
NAND read  |     197               | 197           |      196       |
eMMC read  |     194               | 195           |      196       |
---------------------------------------------------------------------

Because FUSE is a users pace daemon, more threads means it can handle
more request in parallel.

Signed-off-by Zhang Jiejing <jiejing.zhang@freescale.com>
---
 imx6/etc/init.rc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 9b0773c..ceafc92 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -647,7 +647,8 @@ service wifi_mac /system/bin/sh /system/etc/check_wifi_mac.sh
     oneshot
 
 # virtual sdcard daemon running as media_rw (1023)
-service sdcard /system/bin/sdcard /data/media /mnt/shell/emulated 1023 1023
+# default 4 threads have better performance to 2 threads.
+service sdcard /system/bin/sdcard -t4 /data/media /mnt/shell/emulated 1023 1023
     class late_start
 
 # Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
-- 
1.8.0

